dds/DCPS/RTPS/Spdp.cpp

```diff
@@ -2853,6 +2853,12 @@ Spdp::SpdpTransport::choose_recv_socket(ACE_HANDLE h) const
   return unicast_socket_;
 }
 
+bool valid_size(const ACE_INET_Addr& a)
+{
+  const int ipv6_size = 16;
```

Comment:
This doesn't work for IPv6.  On one Linux system I tried, `sizeof(sockaddr_in6)` was 28.  So like I was saying before, this needs to be based on `sizeof` so it agrees with the system's headers.

Reply:
Ok, we have no control over where or how the protected member variable addr_size_ is set, since there are too many functions, such as ACE_Addr::set_size, ACE_Addr::base_set, ACE_INET_Addr::set_address, ACE_INET_Addr::set(). Also any derived class can directly modify addr_size_ (talking about design issues).

ACE_INET_Addr::set() does use the sizeof operator to set addr_size_ which is totally different meaning of addr_size_ as in ACE_INET_Addr::set_address().

ACE_INET_Addr::set()
{
//...
this->base_set (AF_INET6, sizeof this->inet_addr_.in4_);
//...
this->inet_addr_.in6_.sin6_len = sizeof this->inet_addr_.in6_;
//...
this->set_size (sizeof this->inet_addr_.in6_);
//...
}

Based on the above, valid_size is modified to compare addr_size_ with the appropriate sizeof operator.